### ISSUE TITLE
[request] Optimize CI and development process

### ISSUE THREAD

**Issue by apteles** [go-gremlins/gremlins#202]
# 🚀 Feature Request

### Is your feature request related to a problem? Please describe
I'd like to know how we could optimize our pipeline, for example, we always have to do a PR and run all mutation tests
and it spends a lot of time on our pipe. Even when we are coding, we have to execute local tests like unit tests and finally mutation but in general, we coded just in a single module/package or a few of then, and one more once it spent a lot of time until we create our PR.

### Describe the solution you'd like

I saw one solution in php-infection project that they have the following solution:

https://infection.github.io/guide/command-line-options.html#git-diff-filter

In short, it will  **mutate only those files that were added in the Pull Request**. That could be a great feature. I'm sorry if I already have said/suggest a feature that already exists, but the closest that I saw in the documentation was filter by package in the project.

**Comment by k3rn31** [comment]
Hello, thanks for the FR. This seems to be a duplicate of #193.
